### PR TITLE
CEP-78

### PR DESCRIPTION
**Pull request by kpob**
## Summary by CodeRabbit

- **New Features**
  - Added new contract declarations and functionality for CEP-78 standard NFTs, including minting, burning, and transferring tokens.
  - Introduced reverse lookup operations and settings management for various NFT-related configurations.
  - Implemented event structures for NFT actions like Mint, Burn, Transfer, and Approval.
  - Added functionality to deploy and interact with CEP-78 contracts on the live network.
  
- **Enhancements**
  - Updated `Cargo.toml` to include new dependencies enhancing functionality.
  - Improved dictionary storage and contract context operations by accepting byte slices for keys.
  
- **Bug Fixes**
  - Corrected `.gitignore` to exclude system-specific files like `.DS_Store`.

- **Refactor**
  - Streamlined contract container by removing and modifying method signatures and import statements.

- **Chores**
  - Modified the `install-cargo-odra` command in the `justfile` to specify the stable toolchain for installation.

### CODE REVIEW

File: core/src/contract_context.rs [hunk 1]
```diff
@@ -49,7 +49,7 @@ pub trait ContractContext {
     ///
     /// * `dictionary_name` - The name of the dictionary.
     /// * `key` - The key to retrieve the value for.
-    fn get_dictionary_value(&self, dictionary_name: &str, key: &str) -> Option<Bytes>;
```
Comment (kubaplas) [inline core/src/contract_context.rs:52]: Make sure this works by running a livenet example of cep-18 against original contract implementation.

File: core/src/contract_context.rs [hunk 2]
```diff
@@ -58,7 +58,7 @@ pub trait ContractContext {
     /// * `dictionary_name` - The name of the dictionary.
     /// * `key` - The key to set the value for.
     /// * `value` - The value to set.
-    fn set_dictionary_value(&self, dictionary_name: &str, key: &str, value: CLValue);
```
Comment (kubaplas) [inline core/src/contract_context.rs:61]: Why only changed in dictionaries, what about named keys?

**Comment by zie1ony** [comment]
@refraction-autoreview help

File: core/src/contract_env.rs [hunk 3]
```diff
@@ -123,14 +122,13 @@ impl ContractEnv {
     }
 
     /// Sets the value associated with the given named key in the named dictionary in the contract storage.
-    pub fn set_dictionary_value<T: CLTyped + ToBytes, U: AsRef<str>, V: AsRef<str>>(
+    pub fn set_dictionary_value<T: CLTyped + ToBytes, U: AsRef<str>>(
         &self,
         dictionary_name: U,
-        key: V,
+        key: &[u8],
```
[the remark this answers is not in the document]
Comment (kubaplas) [reply core/src/contract_env.rs:128]: DEEP RABBIT

**Review by zie1ony — APPROVED** [review]

**Pull request merged**